spec/timezone.html

```diff
@@ -30,73 +30,76 @@ <h1>Time Zone Names</h1>
         1. Return *false*.
       </emu-alg>
     </emu-clause>
 
-    <emu-clause id="sec-canonicalizetimezonename" aoid="CanonicalizeTimeZoneName">
-      <h1>CanonicalizeTimeZoneName ( _timeZone_ )</h1>
+    <emu-clause id="sec-canonicalizetimezonename" type="abstract operation">
+      <h1>
+        CanonicalizeTimeZoneName (
+          _timeZone_: a String value that is a valid time zone name as verified by IsValidTimeZoneName,
+        )
+      </h1>
+      <dl class="header">
+        <dt>description</dt>
+        <dd>It returns the canonical and case-regularized form of _timeZone_.</dd>
+      </dl>
 
       <p>
         An ECMAScript implementation that includes the ECMA-402 Internationalization API must implement the CanonicalizeTimeZoneName abstract operation as specified in the ECMA-402 specification.
         If an ECMAScript implementation does not include the ECMA-402 API the following specification of the CanonicalizeTimeZoneName abstract operation is used.
       </p>
+
       <p>
-        The abstract operation CanonicalizeTimeZoneName takes argument _timeZone_, a String value, and returns the canonical and case-regularized form of _timeZone_.
-      </p>
-      <p>
-        CanonicalizeTimeZoneName may only be called with a _timeZone_ parameter that is a valid time zone name according to IsValidTimeZoneName.
-      </p>
-      <p>
-        The minimum implementation of CanonicalizeTimeZoneName, supporting only the *"UTC"* time zone, performs the following steps when called:
+        In the descriptions below, String value _A_ is an ASCII-case-insensitive match for String value _B_ if the String value derived from _A_ by replacing each occurrence of a lowercase ASCII letter code unit (0x0061 through 0x007A, inclusive) with the corresponding uppercase ASCII letter code unit (0x0041 through 0x005A, inclusive) while preserving all other code units is exactly the same sequence of code units as the String value that is derived from _B_ in the same way.
       </p>
-
       <emu-alg>
-        1. Assert: Type(_timeZone_) is String.
-        1. Assert: ! IsValidTimeZoneName(_timeZone_) is *true*.
-        1. Return *"UTC"*.
+        1. If _timeZone_ is an ASCII-case-insensitive match for *"UTC"*, return *"UTC"*.
+        1. Return the String value containing the implementation-defined canonical and case-regularized name of the built-in time zone whose name is an ASCII-case-insensitive match for _timeZone_.
       </emu-alg>
+      <emu-note>
+        If _timeZone_ is an ASCII-case-insensitive match for the name of a Link in the IANA Time Zone Database, the value returned from CanonicalizeTimeZoneName(_timeZone_) should be derived from the corresponding Zone name rather than from the Link name. And if either that Zone name or _timeZone_ are *"Etc/UTC"* or *"Etc/GMT"*, the value returned should be *"UTC"*. Both behaviours are required of an ECMAScript implementation that includes the ECMA-402 Internationalization API.
```

Comment:
I'd love to remove _all_ the ECMA-402-specific operation overrides by slightly rewording the already-existing constraints that limit built-in time zone names to those published by IANA and incorporating the Link following into ECMA-262 CanonicalizeTimeZoneName (presumably Normative Optional), but I'm not quite sure how to pull it off. Maybe it will be more clear if IsValidTimeZoneName is removed from both as I suggested above.

Reply:
If we are going in the direction of decoupling them, then this seems no longer applicable.